classes/output/mobile.php

```diff
@@ -0,0 +1,79 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace block_multiblock\output;
+
+/**
+ * Class mobile
+ *
+ * @package    block_multiblock
+ * @copyright  2025 Sumaiya Javed <sumaiya.javed@catalyst.net.nz>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class mobile {
+    /**
+     * Returns the view for the mobile app.
+     *
+     * @param  array $args Arguments from tool_mobile_get_content WS
+     * @return array HTML, javascript and otherdata
+     */
+    public static function mobile_multiblock_view($args) {
```

Comment:
I notice that a lot of what this function does is the same as what `block_multiblock::get_content()` does. The last part is differenet, because it sending the data for mobile rendering while `block_multiblock::get_content()` sends it to the page renderer, but the earlier parts of the function where it retrieves data are pretty similar.

Would it make sense to extract out the data-retrieving part of `block_multiblock::get_content()` and re-use it here?

Also, I'm noticing that `block_multiblock::get_content()` has some stuff in it about "default blocks", and this function does not.

Reply:
It uses different templates and the data structure is also slightly different.
If I use block_multiblock::get_content(), I will to do lot more changes then it is worth.